bgls/utils_test.py

```diff
@@ -0,0 +1,70 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the BGLS Simulator utilities."""
+
+import pytest
+
+import cirq
+
+import bgls
+
+import numpy as np
+
+
+@pytest.mark.parametrize("nqubits", range(3, 5 + 1))
+def test_improved_random_circuit(nqubits: int):
```

Comment:
nit: function name is `generate_random_circuit` now, test ideally should reflect that. Also as a style thing I like to do `test_[function_name]_[does_something_expected]` as a naming convention, so this could be `test_generate_random_circuit_matches_cirq_random_circuit`. It's clearer, especially if you are not adding docstrings to tests, but also more verbose. Up to you!

Reply:
sounds good, will do this for future ones too